### PR TITLE
Mobile fixes

### PR DESCRIPTION
**Pull request by Esinko**
This PR includes 2 (+1 fix) commits. The first of which implements a new navigation system for mobile.

The new navigation system looks like this in it's `closed` state:
image

When opened, the navigation looks like this in its' `open` state:
image
image

The second commit includes changes to the per-project chart. The changes are mostly visual:
image
Changes to note: Legend is now custom, colors are now language specific and the title has been modifed.

### CODE REVIEW

File: src/components/Dashboard.tsx [hunk 1]
```diff
@@ -27,7 +27,7 @@ const useStyles = createStyles(theme => ({
     width: "100%"
   },
   projectCodingChart: {
-    height: "400px",
+    // minHeight: "400px",
```
Comment (Eldemarkki) [inline src/components/Dashboard.tsx:30]: Remove this commented line

File: src/components/PerProjectChart.tsx [hunk 2]
```diff
@@ -87,77 +139,86 @@ export const PerProjectChart = ({ entries, projectCount = 5, className }: PerPro
...
+                    <List.Item key={l}>
+                      {prettifyProgrammingLanguageName(l.slice(0, l.length - 9)) ?? "Unknown"}:{" "}
+                      {prettyDuration(Number(point.data[l]))}
+                    </List.Item>
+                  ))}
+              </List>
+            </Paper>
+          )}
+          valueFormat={v => prettyDuration(v)}
+          layout="horizontal"
```
Comment (Eldemarkki) [inline src/components/PerProjectChart.tsx:197]: Remove this commented line

File: src/index.css [hunk 3]
```diff
@@ -1,4 +1,4 @@
-* {
+*:not(.noDefaultTransition) {
     box-sizing: border-box;
```
Comment (Eldemarkki) [inline src/index.css:2]: I think `box-sizing: border-box` should be applied to the `*` selector. So move this to a new CSS selector block.

**Review by Eldemarkki — CHANGES_REQUESTED** [review]

File: src/components/Dashboard.tsx (hunk 1 above)
Comment (Esinko) [reply src/components/Dashboard.tsx:30]: Oh yeah. I forgot to remove that as I left it there for testing purposes. I had some trouble with the chart scaling.

File: src/index.css (hunk 3 above)
Comment (Esinko) [reply src/index.css:2]: Fair enough, didn't cause any issues in my end though.

File: src/components/PerProjectChart.tsx (hunk 2 above)
Comment (Esinko) [reply src/components/PerProjectChart.tsx:197]: Umm?

Comment (Eldemarkki) [reply src/components/PerProjectChart.tsx:197]: Oops, I think I selected the wrong line. I meant line 198 (`// colors={{ scheme: "paired" }}`)

**Review by Eldemarkki — CHANGES_REQUESTED** [review]

File: src/components/PerProjectChart.tsx (hunk 2 above)
Comment (Esinko) [reply src/components/PerProjectChart.tsx:197]: Oh! Alright, I'll remove that as well.

**Comment by bluelhf** [comment]
this is an excellent pr :)

**Comment by DrVilepis** [comment]
Nice balls!

**Review by Eldemarkki — APPROVED** [review]

**Comment by Eldemarkki** [comment]
Thank you for your contribution! :)

**Pull request merged**